text/0000-version-resource.md

```diff
@@ -0,0 +1,53 @@
+# Version Resource Type
+
+**Status:** `proposed`
+
+Add a composable 'Version' resource type.
+
+## Motivation
+
+When creating trace data or metrics, it can be extremely useful to know the specific version that
+emitted the iota of span or measurement being viewed. However, versions can mean different things
+to different systems and users, leading to a situation where a semantic `Version` type that can
+encapsulate these identifiers would be useful for analysis systems in order to allow for slicing
+data by service, component, or other version.
+
+## Explanation
+
+A `Version` is a semantic resource that can be composed with other resources, such as a `Service`, 
+`Component`, `Library`, `Device`, `Platform`, etc. A `Version` is optional, but recommended.
+The definition of a `Version` is as follows:
+
+| Key     | Value  | Description                                  |
+|---------|--------|----------------------------------------------|
+| semver  | string | version string in semver format              |
+| git_sha | string | version string as a git sha                  |
+
+Only one field should be specified in a single `Version`. 
+
+## Internal details
+
+The exact implementation of this resource would vary based on language, but it would ultimately need to be represented in the data format so that it could be commmunicated to analysis backends. A JSON representation of a version resource follows:
+
+```
+{
+    "version": {
+        "type": "semver",
```

Comment:
Why not `"semver": "1.0.0"`?

Reply:
Is this meant in addition to existing resource objects?
e.g. 
{
	"type": "container",
	"labels": {
           ....
	},
       "version" : {
       ....
       }
}

(example from OpenCensus: https://github.com/census-instrumentation/opencensus-specs/blob/master/resource/Resource.md#exporter-translation)